plugins/de.cognicrypt.codegenerator.tests/src/de/cognicrypt/codegenerator/generator/test/QuestionsToTemplateTests.java

```diff
@@ -0,0 +1,246 @@
+		String actual = wizard.constructTemplateName();
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void testEncryptionTaskWithNoSharingAndByteArray() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		wizard = constructTemplateForEncryptionTask(wizard, "Encryption", "No Sharing", "", "Byte Array", "");
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryption";
+		String actual = wizard.constructTemplateName();
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void testEncryptionTaskWithNoSharingAndFile() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		wizard = constructTemplateForEncryptionTask(wizard, "Encryption", "No Sharing", "", "File", "files");
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionfiles";
+		String actual = wizard.constructTemplateName();
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void testEncryptionTaskWithNoSharingAndString() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		wizard = constructTemplateForEncryptionTask(wizard, "Encryption", "No Sharing", "", "String", "strings");
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionstrings";
+		String actual = wizard.constructTemplateName();
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void testEncryptionTaskWithNoSharingAndOther() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		wizard = constructTemplateForEncryptionTask(wizard, "Encryption", "No Sharing", "", "Other/Do not know", "");
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryption";
+		String actual = wizard.constructTemplateName();
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void testSecurePasswordTask() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		Task securepassword = TestUtils.getTask("SecurePassword");
+		wizard.setSelectedTask(securepassword);
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/securepassword";
+		assertEquals(expected, wizard.constructTemplateName());
+	}
+	
+	@Test
+	public void testDigitalSignaturesTask() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		Task digitalsignatures = TestUtils.getTask("DigitalSignatures");
+		wizard.setSelectedTask(digitalsignatures);
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/digitalsignatures";
+		assertEquals(expected, wizard.constructTemplateName());
+	}
+	
+	private AltConfigWizard constructTemplateForEncryptionTask(AltConfigWizard wizard, String task, String firstAnswer, String firstAnswerOption, String secondAnswer, String secondAnswerOption) {
+		Task encryption = TestUtils.getTask(task);
+		wizard.setSelectedTask(encryption);
+		Question question1 = new Question();
+		question1.setId(0);
+        question1.setQuestionText("Which method of communication would you prefer to use for key exchange?");
```

Comment:
As we discussed before, please do not hard-code any questions here. Instead, read them from the Q&A JSON file of the corresponding task. Then select an answer.
This would also allow this method to become more general purpose than just for the encryption task.